arch/arm/Kconfig

```
@@ -982,6 +982,7 @@ config ARCH_ARMV7R
 	default n
 	select ARCH_HAVE_CPUINFO
 	select ARCH_HAVE_PERF_EVENTS
+	select ARM_THUMB
```

```suggestion
	select ARM_THUMB if BUILD_PROTECTED
```

select ARM_THUMB will cannot be disable by menuconfig, ARM_THUMB should be an option, otherwise it should not exist in kconfig

---

How about we leave it open here then select at processor or even board level?

---

https://github.com/apache/nuttx/blob/master/boards/arm/qemu/qemu-armv7a/configs/knsh/defconfig#L33

---

thumb2 code is default option of cortex-M, not A/R